src/compiler/scanner.ts

```diff
@@ -2159,7 +2159,6 @@ namespace ts {
                         tokenValue = String.fromCharCode(cookedChar) + scanIdentifierParts();
                         return token = getIdentifierToken();
                     }
-                    error(Diagnostics.Invalid_character);
                     pos++;

```

Comment:
Should we also be ignoring errors on invalid Unicode escapes, like `\u{1010101010}`? That'll require passing a flag into `scanIdentifierParts` and the other callers of `scanExtendedUnicodeEscape`.

Reply:
I can't get any error with the current code! I compared

```js
/**
 * unicode-escape = \u{abcdefghi} -- should not have error for invalid unicode escape
 */
var hi = "\u{00fa}\u{abcdefghi}"
```

There's an error for the second `abcdefghi` but not the first.

I'll add a test to make sure it stays that way.

Reply:
You should also check, just in case, the scenario when the unicode escape is the second character in the identifier `eg, q\u{abcdefghi}` - The first character is handled inline in `scanJsDocToken` with a non-erroring `peek` (so it can fallback to the `Invalid character` error you just removed), while the second character (and beyond) is handled in `scanIdentifierParts` which, I could always behave different.

Reply:
Done.